### PR TITLE
Document what's new on the website since Dart 2.16

### PR DESCRIPTION
**Pull request by parlough**
The website has gone through a lot of large changes since the release of Dart 2.16, much much more than usual...

In this PR, I aimed to stick to a format consistent with previous entries and in documenting changes of a similar scope. To ease parsing, I did my best to include the most important changes at the beginning and to collect related changes together. I'm happy to reorganize or cut things that don't seem relevant or important enough.

**Staged:** https://dart-dev--pr4031-feature-2-17-whats-n-66qti5j4.web.app/guides/whats-new#may-11-2022-217-release

### CODE REVIEW

**Review by domesticmouse — APPROVED** [review]

File: src/_guides/whats-new.md [hunk 1]
```diff
@@ -17,6 +17,144 @@ and follow the [Dart blog][].
...
+
+[Dart 2.17: BLOG POST TODO]: https://medium.com/dartlang/TODO
+
+### Docs updated or added to dart.dev
+{: .no_toc}
+
+In addition to bug fixes and incremental improvements,
+we made the following changes to this site:
+
+* Introduced the [Learning Dart as a JavaScript developer][] guide
```
Comment (sfshaza2) [inline src/_guides/whats-new.md:35]:
```suggestion
* Introduced the [Learning Dart as a JavaScript developer][] guide,
```

File: src/_guides/whats-new.md [hunk 2]
```diff
@@ -17,6 +17,144 @@ and follow the [Dart blog][].
...
+  * Updated the linux [installation instructions][get-dart-install]
+    to use [SecureApt][] and follow the latest best practices.
+  * Added support for downloading experimental, Linux RISC-V (RV64GC) builds
+    from the [Dart SDK archive][].
+* Continued work to improve and update documentation
+  of the [unified `dart` tool][dart-tool]:
+  * Expanded documentation about the functionality of the [`dart fix`][] tool.
+  * Adjusted the guidelines and documentation for the [`dart doc`][] tool
+    to match its functionality and underlying behavior.
+  * Added furthered documentation and samples of [`dart compile js`][].
```
Comment (sfshaza2) [inline src/_guides/whats-new.md:66]:
```suggestion
  * Added further documentation and samples of [`dart compile js`][].
```

File: src/_guides/whats-new.md [hunk 3]
```diff
@@ -17,6 +17,144 @@ and follow the [Dart blog][].
...
+  * Removed mentions of the old standalone tools.
```
Comment (sfshaza2) [inline src/_guides/whats-new.md:67]:
```suggestion
  * Removed mentions of the deprecated standalone tools.
```

**Review by sfshaza2 — APPROVED** [review]
VERY nice, @parlough! lgtm with a few minor edits.

File: src/_guides/whats-new.md (hunk 2 above)
Comment (parlough) [reply src/_guides/whats-new.md:66]: Ooops...

File: src/_guides/whats-new.md (hunk 3 above)
Comment (parlough) [reply src/_guides/whats-new.md:67]: The mentions removed are of tools now completely removed. Is deprecated still fine here or should I say something else?

**Review by johnpryan — APPROVED** [review]
Looks great!

**Pull request merged**